### PR TITLE
Revise the notes about registration and gtype of remote datasets

### LINKED ISSUE #2378: Remove the notes that high-resolution remote datasets don't support slice operations
In the documentation of all remote datasets (e.g., https://www.pygmt.org/v0.8.0/api/generated/pygmt.datasets.load_earth_age.html), we have notes like:

> The [xarray.DataArray](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.html#xarray.DataArray) grid doesn’t support slice operation, for Earth seafloor crustal age with resolutions of 5 arc-minutes or higher, which are stored as smaller tiles.

The notes are inaccurate:

1. After PR #2009 was merged, it's OK to do slice operation on high-resolution remote datasets, although the registration and gtype properties are lost (see #2375 and #499 for the reason).
```python
>>> from pygmt.datasets import load_earth_relief
>>> grid = load_earth_relief(resolution="05m", region=[0, 5, 0, 5], registration="pixel")
>>> grid.gmt.registration, grid.gmt.gtype
(1, 1)
>>> grid2 = grid[0:2, 0:2]
>>> grid2.gmt.registration, grid2.gmt.gtype
(0, 0)
```
2. The properties will also be lost for low-resolution datasets after some grid operations:
```python
>>> from pygmt.datasets import load_earth_relief
>>> grid =  load_earth_relief()
>>> grid.gmt.registration, grid.gmt.gtype
(0, 1)
>>> grid.encoding
{'zlib': True,
 'szip': False,
 'zstd': False,
 'bzip2': False,
 'blosc': False,
 'shuffle': True,
 'complevel': 9,
 'fletcher32': False,
 'contiguous': False,
 'chunksizes': (181, 181),
 'source': '/home/seisman/.gmt/server/earth/earth_relief/earth_relief_01d_g.grd',
 'original_shape': (181, 361),
 'dtype': dtype('int16'),
 '_FillValue': -32768,
 'scale_factor': 0.5}
>>> grid2 = grid * 2.0
>>> grid2.encoding
{}
>>> grid2.gmt.registration, grid2.gmt.gtype
(0, 0)
```

These are all caused by the limitations of the xarray accessors (#499), so it should be documented in a single place (as done in PR #2375). So, I think we should remove these notes for remote datasets.

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

See https://github.com/GenericMappingTools/pygmt/issues/2378 for context.

**Preview**: https://pygmt-dev--2384.org.readthedocs.build/en/2384/api/generated/pygmt.datasets.load_earth_age.html

Fixes #2378

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/earth_age.py [hunk 1]
```diff
@@ -54,9 +54,14 @@ def load_earth_age(resolution="01d", region=None, registration=None):
 
     Note
     ----
-    The :class:`xarray.DataArray` grid doesn't support slice operation, for
-    Earth seafloor crustal age with resolutions of 5 arc-minutes or higher,
-    which are stored as smaller tiles.
+
+    The registration and coordinate system type of the returned
+    :class:`xarray.DataArray` grids can be accessed by the GMT accessors
```
Comment (seisman) [inline pygmt/datasets/earth_age.py:59]:
```suggestion
    :class:`xarray.DataArray` grids can be accessed via the GMT accessors
```

File: pygmt/datasets/earth_age.py [hunk 2]
```diff
@@ -54,9 +54,14 @@ def load_earth_age(resolution="01d", region=None, registration=None):
...
-    Earth seafloor crustal age with resolutions of 5 arc-minutes or higher,
-    which are stored as smaller tiles.
+
+    The registration and coordinate system type of the returned
+    :class:`xarray.DataArray` grids can be accessed via the GMT accessors
+    (i.e., ``grid.gmt.registration`` and ``grid.gmt.gtype`` respectively).
+    However, these properties may be lost after specific grid operations and
+    need to be set manually before passing the grid further to PyGMT functions.
+    Refer to :class:`pygmt.GMTDataArrayAccessor` for detailed explanations and
+    workarounds.
```
Comment (weiji14) [inline pygmt/datasets/earth_age.py:64]: Recommendation to still mention slicing as an example of an operation that causes `registration` and `gtype` to be lost, and be explicit that this affects **both** PyGMT [data processing](https://www.pygmt.org/v0.8.0/api/index.html#data-processing) and [plotting](https://www.pygmt.org/v0.8.0/api/index.html#plotting) functions.

```suggestion
    However, these properties may be lost after specific grid operations (such
    as slicing) and will need to be manually set before passing the grid to any
    PyGMT data processing or plotting functions. Refer to
    :class:`pygmt.GMTDataArrayAccessor` for detailed explanations and
    workarounds.
```

File: pygmt/datasets/earth_age.py [hunk 3]
```diff
@@ -54,9 +54,15 @@ def load_earth_age(resolution="01d", region=None, registration=None):
 
     Note
     ----
-    The :class:`xarray.DataArray` grid doesn't support slice operation, for
-    Earth seafloor crustal age with resolutions of 5 arc-minutes or higher,
-    which are stored as smaller tiles.
+
```
Comment (weiji14) [inline pygmt/datasets/earth_age.py:57]: Should there be an extra space under `Note`?
```suggestion
```

**Review by weiji14 — COMMENTED** [review]
@yvonnefroehlich and @michaelgrund, if the wording looks ok, can @seisman apply the changes to the other `datasets/earth_*.py` files?

File: pygmt/datasets/earth_age.py [hunk 4]
```diff
@@ -54,9 +54,14 @@ def load_earth_age(resolution="01d", region=None, registration=None):
...
-    Earth seafloor crustal age with resolutions of 5 arc-minutes or higher,
-    which are stored as smaller tiles.
+    The registration and coordinate system type of the returned
+    :class:`xarray.DataArray` grids can be accessed via the GMT accessors
+    (i.e., ``grid.gmt.registration`` and ``grid.gmt.gtype`` respectively).
+    However, these properties may be lost after specific grid operations (such
+    as slicing) and will need to be manually set before passing the grid to any
+    PyGMT data processing or plotting functions. Refer to
+    :class:`pygmt.GMTDataArrayAccessor` for detailed explanations and
+    workarounds.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_age.py:64]: So far, the wording looks good to me!
Only a tiny comment: I am wondering, whether we should use here singular?
```suggestion
    The registration and coordinate system type of the returned
    :class:`xarray.DataArray` grid can be accessed via the GMT accessors
    (i.e., ``grid.gmt.registration`` and ``grid.gmt.gtype`` respectively).
    However, these properties may be lost after specific grid operations (such
    as slicing) and will need to be manually set before passing the grid to any
    PyGMT data processing or plotting functions. Refer to
    :class:`pygmt.GMTDataArrayAccessor` for detailed explanations and
    workarounds.
```

**Comment by seisman** [comment]
@michaelgrund and @yvonnefroehlich Any comments on this PR? Or can it be approved?

**Review by yvonnefroehlich — APPROVED** [review]

**Pull request merged**